### PR TITLE
Async Filter expression check

### PR DESCRIPTION
**Pull request by AvaniMakwana**
Authored-by: AvaniMakwana

Description
Filter expression check implementation

How Has This Been Tested?
Unit tests have been included.

License
I confirm that this contribution is made under an Apache 2.0 license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: elide-async/src/main/java/com/yahoo/elide/async/models/AsyncQuery.java [hunk 1]
```diff
@@ -64,7 +64,6 @@
     @Embedded
     private AsyncQueryResult result;
 
-    @Exclude
     private String principalName;
```
Comment (moizarafat) [inline elide-async/src/main/java/com/yahoo/elide/async/models/AsyncQuery.java:67]: Lets include `@CreatePermission(expression = "Prefab.Role.None")` on principalName. We don't want it to be set by users, we will be setting it.

File: elide-integration-tests/src/test/java/com/yahoo/elide/async/integration/tests/AsyncIT.java [hunk 2]
```diff
@@ -81,7 +81,7 @@
 
         private String id;
         private String query;
-
+        private String principalName;
```
Comment (aklish) [inline elide-integration-tests/src/test/java/com/yahoo/elide/async/integration/tests/AsyncIT.java:84]: Remove unused variable

File: elide-async/src/main/java/com/yahoo/elide/async/models/security/AsyncQueryFilterExpression.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import java.security.Principal;
+import java.util.Collections;
+
+/**
+ * FilterExpressions for AsyncQuery.
+ */
+public class AsyncQueryFilterExpression {
+    private static final String PRINCIPAL_NAME = "principalName";
+
+    static private FilterPredicate getPredicateOfPrincipalName(String principalName) {
```
Comment (aklish) [inline elide-async/src/main/java/com/yahoo/elide/async/models/security/AsyncQueryFilterExpression.java:26]: Can we consolidate this logic into AsyncQueryInlineChecks.java ?

**Comment by AvaniMakwana** [comment]
updates in https://github.com/yahoo/elide/pull/1505

**Pull request closed without merging**